src/drivers/i2c/i2c.h

```diff
@@ -0,0 +1,37 @@
+#ifndef I2C_H
+#define I2C_H
+
+#include "common.h" // Common definitions
+#include <stdint.h> // Standard integer types
+
+// Supported I2C devices
+typedef enum {
+    I2C_DEVICE_POLULU, // Polulu AltIMU-v6
+    I2C_DEVICE_MOVELLA, // Movella MTi-630 IMU
+    I2C_DEVICE_ST, // ST LSM6DSO32 IMU
+    I2C_DEVICE_COUNT // Number of devices
+} i2c_device_t;
+
+// Initialize an I2C device
+// device: I2C device to initialize
```

Comment:
If we're commenting in this level of detail might as well use doxygen

Reply:
honestly had no idea on what level of detail I should've utilized but would doxygen be good atp if we're assuming the user isn't familiar with the design document? @celery6

Reply:
any code on its own should always contain documentation regardless of whether design doc exists or not. 

Idc about doxygen or any other style tbh but sure lets make doxygen style our standard from now on